frameworks/helloworld/tests/test_mono_to_multi_migrate.py

```diff
@@ -0,0 +1,67 @@
+import logging
+
+import pytest
+from tests import config
+
+import sdk_install
+import sdk_plan
+import sdk_tasks
+import sdk_upgrade
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='function', autouse=True)
+def before_each_test(configure_security):
+    try:
+        sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+        yield  # let the test session execute
+    finally:
+        sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+
+
+@pytest.mark.sanity
+def test_old_tasks_not_relaunched():
+    # Install hello-world with single yaml
+    sdk_install.install(config.PACKAGE_NAME,
+                        config.SERVICE_NAME,
+                        3,
+                        additional_options={"service": {"yaml": "svc"}})
+    hello_task_id = sdk_tasks.get_task_ids(config.SERVICE_NAME, 'hello')
+    # Start update plan with options that have list of yaml files to make it launch in multi service mode
+    sdk_upgrade.update_service(config.PACKAGE_NAME,
+                               config.SERVICE_NAME,
+                               additional_options={"service": {"yamls": "svc,foobar_service_name"}})
+    # Ensure new tasks are launched but the old tasks do not relaunch
+    sdk_plan.wait_for_completed_deployment(config.SERVICE_NAME, multiservice_name='foobar')
+    sdk_tasks.check_task_not_relaunched(config.SERVICE_NAME,
+                                        'hello-0-server',
+                                        hello_task_id.pop(),
+                                        multiservice_name=config.SERVICE_NAME)
+    assert len(sdk_tasks.get_task_ids(config.SERVICE_NAME, 'foo')) == 1
```

Comment:
Hmm, just being more explicit. It would catch the case where the scheduler thinks the deployment is complete but the tasks may not be launched. This PR deals with Mono to Multi Migration and i think its no harm to be explicit in the test.

Reply:
I thought I had already made a comment regarding the use of `hello_task_id.pop()` here:

Two things here:

It may be a good idea to first wait for a completed deployment for both services associated with the multi-service before checking for tasks being relaunched. (update triggers a deployment)
I don't think that hello_task_id.pop() is what we want here:
```
In [3]: foo = ["bar", "baz"]

In [4]: set(foo.pop())
Out[4]: {'a', 'b', 'z'}
```